tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/TinkerHelper.java

```diff
@@ -211,4 +218,52 @@ else if (edgeLabels.length == 1)
     public static Map<Object, Edge> getEdges(final TinkerGraph graph) {
         return graph.edges;
     }
+
+    /**
+     * Search for {@link Property}s attached to {@link Element}s of the supplied {@link ElementType} using the supplied
+     * regex. This is a basic scan+filter operation, not a full text search against an index.
+     */
+    public static <E extends Element> Iterator<Property> search(final TinkerGraph graph, final String regex,
+                                                                final Optional<ElementType> type) {
+
+        final Supplier<Iterator<Element>> vertices = () -> IteratorUtils.cast(graph.vertices());
+        final Supplier<Iterator<Element>> edges = () -> IteratorUtils.cast(graph.edges());
+        final Supplier<Iterator<Element>> vertexProperties =
+                () -> IteratorUtils.flatMap(vertices.get(), v -> IteratorUtils.cast(v.properties()));
+
+        Iterator it;
```

Comment:
Looking at this raises an interesting question: how would an external service provider, who does not have access to the node and edge properties, implement something like this? Assuming maximized efficiency, in the general case we wouldn't want to "always" fetch properties for the input type, on the other hand, if one doesn't don't do, the information to execute the service call may in some cases not be present. 

One option would be to design the interface differently, e.g. expect a map from nodes to their properties as input. Another option could be to tackle the problem architecturally, by providing information about whether or not properties are to be fetched in the service metadata (including optionally to use built-in functionality & interfaces to handle these situations).

I don't have a good answer here as well, but maybe you have any thoughts?